Add Custom Translation Navigation Screen

JoshuaSchaefer opened the pull request:
Add a navigation screen to allow users to navigate between different defined translations. Also, separate adding and editing a custom translation into two different processes.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
The new navigation screen displays buttons that link to each defined translation. Then a separate button to add a new custom translation. When the user adds a new translation, they can select which language they want to define and cancel or save any additions. When a user edits an existing translation, they cannot change the language, but they can delete the translation and cancel or save any changes.

Also, with this change, most of the functionality in the following document is implemented.
[catrobat custom translations](https://docs.google.com/document/d/1IU-12Np7yajbz2Gt9rkfm2jasQlt7trhliu1gqE1k4Y/edit?usp=sharing)

### Tests - additional information
All tests that are impacted by this change have been updated.

dmetzner commented on translations/catroweb.en.yml:
```diff
@@ -205,6 +205,11 @@ programs:
   confrimDelete: Are you sure you want to delete the translation?
   default: Default
   save: Save
+  translationTitle: '%language% Translation'
```
translation*

We try to keep the lower/uppercases grammatically correct.

JoshuaSchaefer replied:
Updated to "translation"

dmetzner commented on translations/catroweb.en.yml:
```diff
@@ -205,6 +205,11 @@ programs:
   default: Default
   save: Save
+  translationTitle: '%language% Translation'
+  editDefault: Edit Default
```
Edit default

JoshuaSchaefer replied:
Updated to "Edit default"

dmetzner commented on translations/catroweb.en.yml:
```diff
@@ -205,6 +205,11 @@ programs:
   save: Save
+  translationTitle: '%language% Translation'
+  editDefault: Edit Default
+  editTranslation: Edit %language% Translation
```
translation

JoshuaSchaefer replied:
Updated to "translation"

dmetzner commented on translations/catroweb.en.yml:
```diff
@@ -205,6 +205,11 @@ programs:
+  editDefault: Edit Default
+  editTranslation: Edit %language% Translation
+  translations: Translations
+  createTranslation: Create Translation
```
translation

JoshuaSchaefer replied:
Updated to "translation"

dmetzner commented on templates/Program/program_editor_navigation.html.twig:
```diff
@@ -0,0 +1,12 @@
+  <div class="col">
+    {% if app.user and my_program %}
+      <div id="edit-text-navigation" class="d-none mb-5 full-screen">
+        <ul id="navigation-language-list" style="list-style-type: none; padding-left: 0px;">
```
If possible move inline styles always to CSS classes. This way they can be optimized and minified.

JoshuaSchaefer replied:
Moved styling to a new CSS class

dmetzner reviewed (approved):
The feature itself looks fine. Maybe consider a max-width on desktop - similar to the navbar.
![image](https://user-images.githubusercontent.com/40868718/159216333-953a0780-ce38-473c-9212-dff9b7064770.png)

Commit:
Add Custom Translation Navigation Screen

Add a navigation screen to allow users to navigate between different defined translations. Also, separate adding and editing a custom translation into two different processes.
